app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,97 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState {
+    class InputText(val text: String) : MovieCategoryLabelTextViewState()
+    class ResourceText(@StringRes val textRes: Int) : MovieCategoryLabelTextViewState()
+}
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+
+)
+
+@Composable
+fun MovieCategoryLabel(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    if (movieCategoryLabelViewState.isSelected) {
+        Column(
+            modifier = Modifier
+                .padding(5.dp)
+                .width(intrinsicSize = IntrinsicSize.Max)
+                .clickable(onClick = onClick)
+        ) {
+            Text(
+                text = selectTextSource(movieCategoryLabelViewState = movieCategoryLabelViewState),
+                color = Color.Black,
+                fontWeight = FontWeight.Bold,
+                fontSize = 16.sp,
+                modifier = Modifier
+                    .fillMaxWidth()
+            )
+            Spacer(
+                modifier = Modifier
+                    .size(6.dp)
+            )
+            Divider(
+                color = Color.Black,
+                thickness = 4.dp,
+                modifier = Modifier
+                    .fillMaxWidth()
+            )
+        }
+    } else {
+        Text(
+            text = selectTextSource(movieCategoryLabelViewState = movieCategoryLabelViewState),
+            color = Color.Gray,
+            fontSize = 16.sp,
+            modifier = modifier
+                .padding(5.dp)
+                .clickable(onClick = onClick)
+        )
+    }
+}
```

Comment:
Your component should have the same height independently of selected state. Currently, if label is not selected, it will display only the text without having divider height included. You can solve this with coloring `Divider` to Transparent color when the labes is not selected. Also, you don't need to call `Text` composable function twice. Instead, you can have one call with different arguments depending on the state of component. Here is the template of what I mean:

```kotlin
@Composable
fun MovieCategoryLabel(
    movieCategoryLabelUiState: MovieCategoryLabelViewState,
    onItemClick: (MovieCategoryLabelViewState) -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .clickable { onItemClick(movieCategoryLabelUiState) }
            // call to other modifiers
    ) {
        Text(
            text = when (movieCategoryLabelUiState.categoryText) {
                is MovieCategoryLabelTextViewState.InputText -> movieCategoryLabelUiState.categoryText.text
                is MovieCategoryLabelTextViewState.ResourceText -> stringResource(movieCategoryLabelUiState.categoryText.textRes)
            },
            color = if (movieCategoryLabelUiState.isSelected) Color.Black else Color.Gray,
            // call to other arguments
        )

        Spacer(
            modifier = Modifier
                .size(6.dp)
            )
          Divider(
              color = if (movieCategoryLabelUiState.isSelected) Color.Black else Color.Transparent,
              thickness = 4.dp,
              modifier = Modifier
                  .fillMaxWidth()
        )
    }
}
```